lib/SILOptimizer/Transforms/CSE.cpp

```diff
@@ -917,19 +1015,24 @@ bool CSE::processNode(DominanceInfoNode *Node) {
         lazyPropertyGetters.push_back(AI);
         continue;
       }
-                              
-      // Instructions producing a new opened archetype need a special handling,
-      // because replacing these instructions may require a replacement
-      // of the opened archetype type operands in some of the uses.
-      if (!isa<OpenExistentialRefInst>(Inst)
-          || processOpenExistentialRef(
-              cast<OpenExistentialRefInst>(Inst),
-              cast<OpenExistentialRefInst>(AvailInst))) {
-        // processOpenExistentialRef may delete instructions other than Inst, so
-        // nextI must be reassigned.
-        nextI = std::next(Inst->getIterator());
-        Inst->replaceAllUsesPairwiseWith(AvailInst);
-        Inst->eraseFromParent();
+      if (!isa<OpenExistentialRefInst>(Inst) ||
+          processOpenExistentialRef(cast<OpenExistentialRefInst>(Inst),
+                                    cast<OpenExistentialRefInst>(AvailInst))) {
+        if (!Inst->getFunction()->hasOwnership() ||
+            Inst->getResults().empty()) {
+          Inst->replaceAllUsesPairwiseWith(AvailInst);
```

Comment:
Can you move no results to a separate case? Seems weird to me call replaceAllUsesPairwiseWith it since it has no results so no uses!